test_vm/tests/change_owner_test.rs

```diff
@@ -1,163 +1,267 @@
+    };
+
+    // when the tests cases are merged, remove the match and use the concrete_vm directly
+    let addrs = create_accounts_(vm, 3, TokenAmount::from_whole(10_000), SignatureType::BLS);
     let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
     let (owner, worker, new_owner, beneficiary) = (addrs[0], addrs[0], addrs[1], addrs[2]);
 
     // create miner
-    let miner_id = create_miner(
-        &mut v,
+    let miner_id = create_miner_(
+        vm,
         owner,
         worker,
         seal_proof.registered_window_post_proof().unwrap(),
         TokenAmount::from_whole(1_000),
     )
     .0;
 
-    change_beneficiary(
-        &v,
+    change_beneficiary_(
+        vm,
         owner,
         miner_id,
         &ChangeBeneficiaryParams::new(beneficiary, TokenAmount::from_atto(100), 100),
     );
-    change_owner_address(&v, owner, miner_id, new_owner);
-    let miner_info = v.get_miner_info(miner_id);
-    assert_eq!(new_owner, miner_info.pending_owner_address.unwrap());
+    change_owner_address_(vm, owner, miner_id, new_owner);
+    match concrete_vm {
+        ConcreteVM::TestVM(tvm) => {
+            let miner_info = tvm.get_miner_info(miner_id);
+            assert_eq!(new_owner, miner_info.pending_owner_address.unwrap());
+        }
+        #[cfg(feature = "benchmark")]
+        ConcreteVM::_BenchmarkVM(_) => {
+            // this block shouldn't compile unless the feature is enabled
+            compilation_error;
+        }
+        _ => {}
+    }
 
-    change_owner_address(&v, new_owner, miner_id, new_owner);
-    let miner_info = v.get_miner_info(miner_id);
-    assert!(miner_info.pending_owner_address.is_none());
-    assert_eq!(new_owner, miner_info.owner);
-    assert_eq!(new_owner, miner_info.beneficiary);
+    change_owner_address_(vm, new_owner, miner_id, new_owner);
 
-    v.assert_state_invariants();
+    match concrete_vm {
+        ConcreteVM::TestVM(tvm) => {
+            let miner_info = tvm.get_miner_info(miner_id);
+            assert!(miner_info.pending_owner_address.is_none());
+            assert_eq!(new_owner, miner_info.owner);
+            assert_eq!(new_owner, miner_info.beneficiary);
+        }
+        #[cfg(feature = "benchmark")]
+        ConcreteVM::_BenchmarkVM(_) => {
+            // this block shouldn't compile unless the feature is enabled
+            compilation_error;
+        }
+        _ => {}
+    }
 }
 
+// TODO: this would be gated by #[cfg(feature = "benchmark")] to prevent it running/compiling normally
+// #[cfg(feature = "benchmark")]
 #[test]
-fn keep_beneficiary_when_owner_changed() {
-    let store = MemoryBlockstore::new();
-    let mut v = VM::new_with_singletons(&store);
-    let addrs = create_accounts(&v, 3, TokenAmount::from_whole(10_000));
-    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
-    let (owner, worker, new_owner, beneficiary) = (addrs[0], addrs[0], addrs[1], addrs[2]);
+fn benchmark_change_owner_success() {
```

Comment:
Omit this here. We'll add benchmark entry points later (probably in another repo)